lib/request-proxy/send.js

```diff
@@ -223,19 +223,35 @@ RequestProxySend.prototype.scheduleRetry = function scheduleRetry(callback) {
 RequestProxySend.prototype.send = function send(channelOpts, callback) {
     var self = this;
 
-    this.ringpop.channel.send(
-        channelOpts,
+    channelOpts.service = 'ringpop';
+
+    this.ringpop.channel.request(
+        _.extend({}, channelOpts)
+    ).send(
         channelOpts.endpoint,
         this.getStrHead(),
         this.request.body,
-        this.maxRetries === 0 ? callback : onSend
+        onSend
     );
 
     this.ringpop.emit('requestProxy.requestProxied');
 
-    function onSend(err, res1, res2) {
+    function onSend(err, res, arg2, arg3) {
+        if (!err && res && !res.ok) {
```

Comment:
here we check whether `res` actually exists. Other places, we don't. What is true?